packages/runtime-core/__tests__/vnode.spec.ts

```diff
@@ -1,15 +1,49 @@
-describe('vnode', () => {
-  test.todo('create with just tag')
-
-  test.todo('create with tag and props')
-
-  test.todo('create with tag, props and children')
+import { createVNode } from '@vue/runtime-test'
 
-  test.todo('create with 0 as props')
-
-  test.todo('class normalization')
-
-  test.todo('style normalization')
+describe('vnode', () => {
+  test('create with just tag', () => {
+    const vnode = createVNode('p')
+    expect(vnode.type).toBe('p')
+    expect(vnode.props).toBe(null)
+  });
+
+  test('create with tag and props', () => {
+    const vnode = createVNode('p', {})
+    expect(vnode.type).toBe('p')
+    expect(vnode.props).toMatchObject({})
+  })
+
+  test('create with tag, props and children', () => {
+    const vnode = createVNode('p', {}, ['foo'])
+    expect(vnode.type).toBe('p')
+    expect(vnode.props).toMatchObject({})
+    expect(vnode.children).toMatchObject(["foo"])
+  })
+
+  test('create with 0 as props', () => {
+    const vnode = createVNode('p', null)
+    expect(vnode.type).toBe('p')
+    expect(vnode.props).toBe(null)
+  })
+
+  test('class normalization', () => {
+    let vnode = createVNode('p', { class: 'foo baz'})
+    expect(vnode.props).toMatchObject({ class: 'foo baz'})
+
+    vnode = createVNode('p', { class: ['foo', 'baz']})
+    expect(vnode.props).toMatchObject({ class: 'foo baz'})
+
+    vnode = createVNode('p', { class: {foo: 'foo', baz: 'baz'}})
+    expect(vnode.props).toMatchObject({ class: 'foo baz'})
+  })
+
+  test('style normalization', () => {
+    let vnode = createVNode('p', { style: [{ foo: 'foo'}, { baz: 'baz'}]})
+    expect(vnode.props).toMatchObject({ style: { foo: 'foo', baz: 'baz'}})
+
+    vnode = createVNode('p', { style: {foo: 'foo', baz: 'baz'}})
```

Comment:
Please breakdown this test into multiple tests.

Reply:
done